core/src/main/scala/com/gu/AppIdentity.scala

```diff
@@ -13,30 +15,28 @@ import scala.util.{Failure, Success, Try}
 sealed trait AppIdentity
 
 case class AwsIdentity(
-                        app: String,
-                        stack: String,
-                        stage: String,
-                        region: String
-                      ) extends AppIdentity
+  app: String,
+  stack: String,
+  stage: String,
+  region: String
+) extends AppIdentity
 
-case class DevIdentity(
-                        app: String
-                      ) extends AppIdentity
+case class DevIdentity(app: String) extends AppIdentity
 
 object AppIdentity {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private def safeAwsOperation[A](errorMessage: => String)(operation: => A): Option[A] = Try(operation) match {
-    case Success(value) => Option(value)
-    case Failure(e) =>
-      logger.error(errorMessage, e)
-      None
-  }
+  private def safeAwsOperation[A](errorMessage: => String)(operation: => A): Try[A] =
+    Try(operation).recoverWith {
+      case e: Throwable =>
+        logger.error(errorMessage, e)
+        Failure(e)
+    }
 
-  private def fromASGTags(credentials: => AwsCredentialsProvider): Option[AwsIdentity] = {
+  private def fromASGTags(credentials: => AwsCredentialsProvider): Try[AwsIdentity] = {
```

Comment:
now this will return a Try, which would contain a suitable message if it's run in DEV, to draw attention to the fact it had to time out before it realised it's in DEV